src/lib/url-utils.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * Extracts the hostname from a URL string.
+ * @param url The URL string to extract the hostname from
+ * @returns The hostname (domain and port if specified), or an empty string if the URL is invalid
+ */
+export function getHostnameFromUrl( url: string ): string {
+	try {
+		const parsedUrl = new URL( url );
+		return parsedUrl.hostname;
+	} catch ( error ) {
+		return '';
```

Comment:
@epeicher , should we return the whole `url` param if we cannot parse it?

Reply:
Thinking again about it, I think it is a better UX than displaying an empty string. Let me create a quick PR for that.